src/app/core/services/unit.service.ts

```diff
@@ -5,43 +5,40 @@ import { Injectable } from '@angular/core';
 
 declare const window: any;
 
+type TempUnit = Omit<UnitDetails, 'stats' | 'ability' | 'target' | 'pattern' | 'special'>;
+
 // dont declare service as singleton (i.e., providedIn: 'root')
 // so that each client of the service can get an instance based on its needs
 // (eg, instance per module or instance per component)
 @Injectable()
 class UserService {
 
-  private rumbleUnits: rumble.Unit[] = [];
+  private rumbleUnits: UnitDetails[] = [];
 
   constructor() {
-    let unitDetail: UnitDetails;
-    const unitDetails: UnitDetails[] = [];
-    let unit: rumble.Unit;
-    const units: rumble.Entry[] = rumbleData.units as rumble.Entry[];
-    let vsUnit = false;
-
+    const unitDetails: TempUnit[] = [];
     for (let i = 0; i < window.units.length; i++){
-      if (window.units[i].incomplete ){
+      if (window.units[i].incomplete) {
           continue;
       }
-      vsUnit = false;
-      unitDetail = new UnitDetails();
-      unitDetail.id = i + 1;
-      unitDetail.complete = !window.units[i].incomplete;
-      if (!unitDetail.complete) {
-        unitDetails.push(unitDetail);
-        continue;
-      }
-      unitDetail.isBaseForm = window.evolutions[i + 1];
-      unitDetail.name = window.units[i][0];
-      unitDetail.baseHp = window.units[i][12];
-      unitDetail.baseAtk = window.units[i][13];
-      unitDetail.baseRcv = window.units[i][14];
+
+      let vsUnit = false;
+      const id = i + 1;
+      const unitDetail: TempUnit = {
+        id,
+        complete: true,
```

Comment:
Are we using complete at all at this point? is it always true?

Reply:
yeah the "complete" concept goes away when we decided to stick with the above `if (window.units[i].incomplete) { continue; }`

Reply:
ok, we'll remove this in the future then.